libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -305,7 +308,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 	@Override
 	public void finish() {
         initAnalyticsManager();
-        SalesforceSDKManager.getInstance().getUserAccountManager().sendUserSwitchIntent();
+        final UserAccountManager userAccountManager = SalesforceSDKManager.getInstance().getUserAccountManager();
```

Comment:
Let me know if this logic should be moved elsewhere. `getAuthenticatedUsers()` sometimes takes a long time to run, in my experience.